plannerbenchmark/postProcessing/metrics.py

```diff
@@ -40,42 +62,67 @@ def computeMetric(self, data):
+
+    Requires the threshold, `des_distance`.
+    """
+
     def computeMetric(self, data):
-        m = self._params['m']
+        m = self._params["m"]
         fks = np.stack([data[name] for name in self._measNames[:m]]).T
         goal = np.stack([data[name] for name in self._measNames[m:-1]]).T
         t = data[self._measNames[-1]]
         des_distance = self._params["des_distance"]
         distances = computeDistances(fks, goal)
-        minDistance = np.min(distances)
         indices = np.where(distances < des_distance)
         if indices[0].size == 0:
-            return {'short': -1, 'flag': -1}
+            return {"short": -1, "flag": -1}
         else:
-            return {'short': float(t[np.min(indices)]), 'flag': 0}
+            return {"short": float(t[np.min(indices)]), "flag": 0}
 
 
 class IntegratedErrorMetric(Metric):
+    """Metric to compute the integrated deviation error from reference trajectory.
+
+    Requires the threshold, `des_distance`.
+    This metric computes the averaged deviation error from the first time the
+    threshold was reached.
+    """
+
     def computeMetric(self, data):
-        m = self._params['m']
-        des_distance = self._params['des_distance']
+        m = self._params["m"]
+        des_distance = self._params["des_distance"]
         fks = np.stack([data[name] for name in self._measNames[:m]]).T
         goal = np.stack([data[name] for name in self._measNames[m:-1]]).T
         t = np.array(data[self._measNames[-1]])
         distances = computeDistances(fks, goal)
         indices = np.where(distances < des_distance)
         if indices[0].size == 0:
-            return {'short': 1000}
+            return {"short": 1000}
         else:
-            trackingTime = float((t[-1] - t[indices[0][0]]))
-            trackingError = float(np.sum(distances[indices[0][0]:]))/trackingTime
-            return {'short': trackingError}
+            # trackingTime = float((t[-1] - t[indices[0][0]]))
+            # trackingError = float(np.sum(distances[indices[0][0]:]))/trackingTime
+            trackingError = np.average(distances[indices[0][0] :])
+            return {"short": trackingError}
 
 
 class ClearanceMetric(Metric):
+    """Metric to compute the minimum clearance from any obstacle.
+
+    Requires the dimension of obstacles, `m`, the dimension
+    of the configuration space, `n`,
+    all obstacles present in the scenario, `obstacles` and
+    the link inflation `r_body`.
+    Based on the forward kinematics, the distance between all links and all
+    obstacles is computed. The clearance is the minumum of all those values.
+    The output of this metric provides information about the minimum
+    distance between every link and every obstacle.
+
+    """
+
     def computeMetric(self, data):
         obstacles = self._params["obstacles"]
-        m = self._params["m"]
+        m = obstacles[0].dim()
         n = self._params["n"]
```

Comment:
The variable 'n' can be given a more representative name. It appears in multiple other places such as exp.yml and experiment.py

Reply:
Separate issue for that. #19